internal/repository/employee/employee.go

```diff
@@ -2,91 +2,108 @@ package employee
 
 import (
 	"context"
+	"database/sql"
 	"errors"
-	"fmt"
-	"sync"
-
 	"github.com/dilyara4949/employees-api/internal/domain"
-
-	"github.com/google/uuid"
 )
 
 type PositionsRepository interface {
 	Get(ctx context.Context, id string) (*domain.Position, error)
 }
 
 type employeeRepository struct {
-	mu            sync.RWMutex
-	storage       map[string]domain.Employee
+	db            *sql.DB
 	positionsRepo PositionsRepository
 }
 
-func NewEmployeesRepository(positionsRepo PositionsRepository) domain.EmployeesRepository {
+func NewEmployeesRepository(db *sql.DB, positionsRepo PositionsRepository) domain.EmployeesRepository {
 	return &employeeRepository{
-		storage:       make(map[string]domain.Employee),
+		db:            db,
 		positionsRepo: positionsRepo,
 	}
 }
 
+var (
+	ErrEmployeeNotFound = errors.New("employee not found")
+	ErrNothingChanged   = errors.New("nothing changed")
+)
+
 func (e *employeeRepository) Create(ctx context.Context, employee *domain.Employee) error {
 	if _, err := e.positionsRepo.Get(ctx, employee.PositionID); err != nil {
-		return fmt.Errorf("error to create employee: %w", err)
+		return err
 	}
 
-	employee.ID = uuid.New().String()
-
-	e.mu.Lock()
-	defer e.mu.Unlock()
-	e.storage[employee.ID] = *employee
+	stmt := "insert into employees (id, first_name, last_name, position_id, created_at) values ($1, $2, $3, $4, CURRENT_TIMESTAMP);"
 
+	if _, err := e.db.Exec(stmt, employee.ID, employee.FirstName, employee.LastName, employee.PositionID); err != nil {
+		return err
+	}
 	return nil
 }
 
 func (e *employeeRepository) Get(_ context.Context, id string) (*domain.Employee, error) {
-	e.mu.RLock()
-	defer e.mu.RUnlock()
-
-	if employee, ok := e.storage[id]; ok {
+	stmt := "select first_name, last_name, position_id from employees where id = $1;"
+	row := e.db.QueryRow(stmt, id)
```

Comment:
As I mentioned before, you should utilize contexts when you work with DB most of the times as they allow you to cancel requests or handle stuck requests. Consider using `QueryRowContext` instead of `QueryRow`. You also would need properly set timeout on a context.

Reply:
fixed